app/assets/javascripts/discourse/app/helpers/category-link.js

```diff
@@ -54,8 +54,11 @@ export function categoryBadgeHTML(category, opts) {
   const depth = (opts.depth || 1) + 1;
   if (opts.recursive && depth <= siteSettings.max_category_nesting) {
     const parentCategory = Category.findById(category.parent_category_id);
+    const lastSubcategory = !opts.depth;
```

Comment:
Why not set `opts` directly?

```suggestion
    opts.lastSubcategory = !opts.depth;
```

Reply:
There is a recursive call to `categoryBadgeHTML` a few lines down. We want the `lastSubcategory` flag to be set only for the last call.